resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/wav/WavTrackImpl.java

```diff
@@ -209,9 +188,65 @@ private void getFormat(InputStream stream) throws IOException {
             frameSize = (int) (period * format.getChannels() * format.getSampleSize() * format.getSampleRate() / 8000);
             duration = sizeOfData * period * 1000000L / frameSize;
         }
+        
+        rbc.close();
+        //close();
+    }
+    
+    public void setPeriod(int period) {
+        this.period = period;
+        frameSize = (int) (period * format.getChannels() * format.getSampleSize() * format.getSampleRate() / 8000);
+    }
+
+    public int getPeriod() {
+        return period;
+    }
+    
+    @Override
+    public Format getFormat() {
+        return format;
+    }
+
+    @Override
+    public long getMediaTime() {
+        return 0;
     }
 
-    /**
+    @Override
+    public void setMediaTime(long timestamp) {
+        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    }
+
+    @Override
+    public long getDuration() {
+        return duration;
+    }
+    
+    // check the channel
+    private void skip(long timestamp) {
```

Comment:
can we reuse this on https://github.com/atika2k/mediaserver/blob/be6bf8f98325f69bd8ba1b265a7705c2d35f1a49/resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/wav/WavTrackImpl.java#L169?